### ISSUE TITLE
Add support for weighted loss functions

### ISSUE THREAD

**Issue by nkaenzig** [kaiko-ai/eva#684]
For imbalanced datasets adding class weights to the loss function can be important. However, as for now for instance the `weight` parameter in `CrossEntropyLoss` can't be use, because it expects a `torch.Tensor` which cannot be specified in the .yaml config / initialized by jsonargparse.
